### PR TITLE
Created an Update button to save all data from ProfileEdit page and send it to backend

### PR DESCRIPTION
**Pull request by docia**
- [x] removed **Back to your personal profile** button and implemented **Update** button instead. 
- [x] the **Update** button is **disabled** until any data is changed.
- [x] on click on the Update button, the changed data is **saved and sent** to the backend. 
- [x] created an alert for a successful update of the profile data with @YulichkaR 's help: [PR](https://github.com/ita-social-projects/SpaceToStudy-Client/pull/2350)
- [x] tests maintained 
Screenshot 2024-07-31 at 15 21 21

-----
The **data which is not saved yet**: 

- _photo_
- _mainSubjects ( aka categories )_
- _videolink_

as the following tasks should be done first: [mainSubjects update](https://github.com/ita-social-projects/SpaceToStudy-Client/issues/2260), [photo update](https://github.com/ita-social-projects/SpaceToStudy-Client/issues/2261), [videolink update](https://github.com/ita-social-projects/SpaceToStudy-Client/issues/2262)

TODO comments are in EditProfile.tsx

-----
### How it works now:

https://github.com/user-attachments/assets/9b495f0c-1393-4d60-8ad7-0683682921f0

---

**PS.** you may notice that after I click Update button, the page is refreshed. Please ignore it; it won't be refreshed on your PC in this way. I refresh the page on purpose because there are local issues on my PC.

### CODE REVIEW

**Review by Olenka-Hryk — APPROVED** [review]

**Comment by Olenka-Hryk** [comment]
Excellent work 🔥

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "allotment": "^1.19.3",
     "axios": "^1.6.0",
     "date-fns": "^2.30.0",
+    "deep-object-diff": "^1.1.9",
```
Comment (YaroslavChuiko) [inline package.json:23]: Do we really need this new dependency to implement this functionality?

File: src/containers/edit-profile/notification-tab/NotificationTab.tsx
Comment (YaroslavChuiko) [inline src/containers/edit-profile/notification-tab/NotificationTab.tsx:1]: I got these errors when trying to open the Notification tab

image

image

**Review by YaroslavChuiko — COMMENTED** [review]
I can update my profile data even if I have errored field

image

image

Also, I think it will be a good idea to notify the user that the data was successfully updated using this component for example 
image

File: package.json (hunk 1 above)
Comment (docia) [reply package.json:23]: yes. I am using diff() function to compare the initial state with the current state to disable or unable the Update button: 
Screenshot 2024-08-14 at 22 38 35

File: src/containers/edit-profile/notification-tab/NotificationTab.tsx
Comment (docia) [reply src/containers/edit-profile/notification-tab/NotificationTab.tsx:1]: there will be no error if you run the backend on this branch https://github.com/ita-social-projects/SpaceToStudy-BackEnd/pull/844

**Comment by docia** [comment]
> I can update my profile data even if I have errored field
> 
> image
> 
> image
> 
> Also, I think it will be a good idea to notify the user that the data was successfully updated using this component for example image

1. the fields with errors will no longer be saved. 
2. i will create another task for this alert since I believe i won't be able to reproduce it with the dispatch issues on my PC

File: package.json (hunk 1 above)
Comment (docia) [reply package.json]: removed diff() and created the following function:
photo_2024-08-16_14-56-57

**Review by YaroslavChuiko — APPROVED** [review]

**Review by YulichkaR — APPROVED** [review]

**Review by Pasichnichenkoilya — APPROVED** [review]

**Review by dudchakk — APPROVED** [review]
amazing work!! 🥰

**Pull request merged**